modules/runtime/viewer/pygame_viewer.py

```diff
@@ -57,85 +62,177 @@ def drawMap(self, map):
+                   dashed=False, zorder=10, linewidth=1):
+        transformed_lines = self.pointsToCameraCoordinate(line2d)
+        if 0 < alpha < 1:
+            s = self.createTransparentSurace(
+                self.screen_dims, self.background_color, alpha)
+
+            if dashed:
+                self.drawDashedLines(
+                    s, self.getColor(color), transformed_lines, 3)
+            else:
+                pg.draw.lines(
+                    s,
+                    self.getColor(color),
+                    False,
+                    transformed_lines,
+                    linewidth)
+        elif alpha == 1:
+            pg.draw.lines(self.screen_surface, self.getColor(
+                color), False, transformed_lines, linewidth)
+
+    def drawPolygon2d(self, polygon, color="blue", alpha=1.0, facecolor=None):
+        transformed_points = self.pointsToCameraCoordinate(polygon)
+        if 0 < alpha < 1:
+            s = self.createTransparentSurace(
+                self.screen_dims, self.background_color, alpha)
+            pg.draw.polygon(s, self.getColor(color), transformed_points)
+        elif alpha == 1:
+            pg.draw.polygon(
+                self.screen_surface,
+                self.getColor(color),
+                transformed_points)
 
     def drawTrajectory(self, trajectory, color):
         if len(trajectory) < 1:
             return
         point_list = []
         for state in trajectory:
-            point_list.append([state[round(StateDefinition.X_POSITION)], state[round(StateDefinition.Y_POSITION)]])
+            point_list.append([state[round(StateDefinition.X_POSITION)],
+                               state[round(StateDefinition.Y_POSITION)]])
 
-        pg.draw.lines(self.screen_surface, self.getColor(color), False, self.pointsToCameraCoordinate(point_list), 5)
+        pg.draw.lines(self.screen_surface, self.getColor(color), False,
+                      self.pointsToCameraCoordinate(point_list), 5)
 
     def drawText(self, position, text, **kwargs):
         font = pg.font.get_default_font()
-        fontsize = kwargs.pop("fontsize", 18)
+        font_size = kwargs.pop("fontsize", 18)
         color = kwargs.pop("color", (0, 0, 0))
-        background_color = kwargs.pop("background_color", (255, 255, 255))
-        text_surface = pg.font.SysFont(font, fontsize).render(text, True, color, background_color)
-        self.screen_surface.blit(text_surface, (position[0] * self.screen_width, position[1] * self.screen_height))
+        background_color = kwargs.pop(
+            "background_color", self.background_color)
+        text_surf = pg.font.SysFont(font, font_size).render(
+            text, True, color, background_color)
+        self.screen_surface.blit(
+            text_surf, (position[0] * self.screen_width, (1 - position[1]) * self.screen_height))
 
     def getColor(self, color):
-        if isinstance(color, Viewer.Color):
+        if isinstance(color, str):
```

Comment:
why did you make these changes regarding color?

Reply:
It was a hard fix to the original code, it could never hit the if case with Viewer.Color, and raise error because once into the else.
For example, somewhere in bark set the color as string "blue".

Reply:
Should I look into the issue?